infrastructures/docker/environment/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/dockerimage/EntryPointParser.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment.dockerimage;
+
+import static com.fasterxml.jackson.databind.node.JsonNodeType.STRING;
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.che.api.core.model.workspace.config.MachineConfig;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Can be used to parse container entry-point definition specified as a YAML list of strings or a
+ * single YAML string.
+ */
+public final class EntryPointParser {
+  private final YAMLMapper mapper = new YAMLMapper();
+
+  /**
+   * Parses the attributes contained in the provided machine config and produces an entry point
+   * definition.
+   *
+   * @param machineAttributes the attributes of a machine to extract the entry point info from
+   * @return an entry point definition, never null
+   * @throws InfrastructureException on failure to parse the command or arguments
+   */
+  public EntryPoint parse(Map<String, String> machineAttributes) throws InfrastructureException {
+    String command = machineAttributes.get(MachineConfig.CONTAINER_COMMAND_ATTRIBUTE);
+    String args = machineAttributes.get(MachineConfig.CONTAINER_ARGS_ATTRIBUTE);
+
+    List<String> commandList =
+        command == null
+            ? emptyList()
+            : parseAsList(command, MachineConfig.CONTAINER_COMMAND_ATTRIBUTE);
+    List<String> argList =
+        args == null ? emptyList() : parseAsList(args, MachineConfig.CONTAINER_ARGS_ATTRIBUTE);
+
+    return new EntryPoint(commandList, argList);
+  }
+
+  /**
+   * Parses a single entry in the format expected by the entry point attribute, i.e. a YAML list of
+   * strings or a YAML string.
+   *
+   * @param data the string data to parse
+   * @return a parsed list or null if the data is not a YAML list or YAML string
+   * @throws IllegalArgumentException if the parsing of the {@literal entry} failed for other
+   *     reasons
+   */
+  public @Nullable List<String> parseEntry(String data) throws IllegalArgumentException {
+    try {
+      JsonNode node = mapper.reader().readTree(data);
+      switch (node.getNodeType()) {
+        case ARRAY:
+          List<String> ret = new ArrayList<>();
+          for (JsonNode n : node) {
+            if (n.getNodeType() != STRING) {
+              return null;
+            }
+            ret.add(n.textValue());
+          }
+          return ret;
+        case STRING:
+          return singletonList(node.textValue());
+        default:
+          return null;
```

Comment:
Maybe it would be better to throw an exception here instead of returning null?

Reply:
Yeah, I admit to being lazy here :wink: